javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+    public Object firstValue_Java(BenchmarkState state) {
+        return state.javaTreeMap.firstKey();
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object firstValue_Scala(BenchmarkState state) {
+        return state.scalaTreeMap.firstKey();
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lastValue_Slang(BenchmarkState state) {
+        return state.slangTreeMap.max();
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lastValue_Java(BenchmarkState state) {
+        return state.javaTreeMap.lastKey();
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lastValue_Scala(BenchmarkState state) {
+        return state.scalaTreeMap.lastKey();
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object floor_Slang(BenchmarkState state) {
+        return state.slangTreeMap.floor(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object floor_Java(BenchmarkState state) {
+        return state.javaTreeMap.floorEntry(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object ceiling_Slang(BenchmarkState state) {
+        return state.slangTreeMap.ceiling(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object ceiling_Java(BenchmarkState state) {
+        return state.javaTreeMap.ceilingEntry(state.nextRandomKey());
+    }
+    @Benchmark
+    @Group("Navigate")
+    public Object higher_Slang(BenchmarkState state) {
+        return state.slangTreeMap.higher(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object higher_Java(BenchmarkState state) {
+        return state.javaTreeMap.higherEntry(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lower_Slang(BenchmarkState state) {
+        return state.slangTreeMap.lower(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lower_Java(BenchmarkState state) {
+        return state.javaTreeMap.lowerEntry(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Slang(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Tuple2<Integer, String> entry : state.slangTreeMap) {
+            values.push(entry._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Java(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Map.Entry<Integer, String> aTreeMap : state.javaTreeMap.entrySet()) {
+            values.push(aTreeMap.getKey());
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Scala(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        Iterator<scala.Tuple2<Integer, String>> iterator = state.scalaTreeMap.iterator();
```

Comment:
you should be able to make a for loop out of this one also, I think :/ (for consistency)